LML.NPOManagement.Bll/Services/UserService.cs

```diff
@@ -263,5 +263,44 @@ public async Task<bool> ModifyUserInfo(UserInformationModel userInformationModel
             await _dbContext.SaveChangesAsync();
             return true;
         }
+
+        public async Task<List<UserInformationModel>> GetUserByUsername(string userName, bool showGroupsOnly, int userId)
+        {
+            List<User> users = new();
+
+            if (!showGroupsOnly)
+            {
+                users = await _dbContext.Users.ToListAsync();
```

Comment:
What if you have 10,000,000 users? Will it return all of them? This can be a very time-consuming operation and takes a lot of memory. Think about whether you really need to get all the users here.